### PR TITLE
feat(theme-classic): new configuration syntax for a simple footer

### LINKED ISSUE #6101: Simple version of footer
### Have you read the Contributing Guidelines on issues?

- [X] I have read the [Contributing Guidelines on issues](https://github.com/facebook/docusaurus/blob/main/CONTRIBUTING.md#reporting-new-issues).

### Motivation

https://infima.dev/docs/components/footer

Currently only one of version of footer is supported, and for project that do not want/have many links, simplified version can be useful

image
image

### Self-service

- [ ] I'd be willing to do some initial work on this proposal myself.

### PR DESCRIPTION
**Pull request by christopherklint97**
## Motivation

Fixes #6101 

### Have you read the [Contributing Guidelines on pull requests](https://github.com/facebook/docusaurus/blob/main/CONTRIBUTING.md#pull-requests)?

Yes.

## Test Plan

I actually had some trouble verifying that the simple footer works, but the multi-column footer still seems to work when viewing `localhost:3000` without changing anything in `docusaurus.config.js`.

My trouble with simple footer was that I got the following error from `yarn start` when I changed the footer in `docusaurus.config.js` to only link items without titles:
```sh
A validation error occurred.
The validation system was added recently to Docusaurus as an attempt to avoid user configuration errors.
We may have made some mistakes.
If you think your configuration is valid and should keep working, please open a bug report.

ValidationError: "footer.links[0].label" is not allowed
```
It seems like this validation error is being thrown here: https://github.com/facebook/docusaurus/blob/main/packages/docusaurus-utils-validation/src/validationUtils.ts#L92

This is what I changed the footer to in `docusaurus.config.js` https://github.com/facebook/docusaurus/blob/main/website/docusaurus.config.js#L418:
```js
footer: {
  style: 'dark',
  links: [
    {
      label: 'Privacy',
      href: 'https://opensource.facebook.com/legal/privacy/',
    },
    {
      label: 'Terms',
      href: 'https://opensource.facebook.com/legal/terms/',
    },
    {
      label: 'Data Policy',
      href: 'https://opensource.facebook.com/legal/data-policy/',
    },
    {
      label: 'Cookie Policy',
      href: 'https://opensource.facebook.com/legal/cookie-policy/',
    },
  ],
  logo: {
    alt: 'Facebook Open Source Logo',
    src: 'img/oss_logo.png',
    width: 160,
    height: 51,
    href: 'https://opensource.facebook.com',
  },
  copyright: `Copyright © ${new Date().getFullYear()} Facebook, Inc. Built with Docusaurus.`
```

Maybe some cache I am missing to clear or some Joi validation I need to change?

### CODE REVIEW

**Comment by Josh-Cena** [comment]
Indeed, you would need to add validation here: https://github.com/facebook/docusaurus/blob/main/packages/docusaurus-theme-classic/src/validateThemeConfig.ts

Joi is quite painful and the current PR is already a good start. Do you want to wrap it up (docs + validation) or would you prefer me to take it over?

**Comment by christopherklint97** [comment]
Awesome, thanks for the help! Now it works great 💪🏾 
Screenshot 2021-12-20 093248
Screenshot 2021-12-20 093315

**Comment by Josh-Cena** [comment]
Cool. Didn't know if Joi schema like that actually works—would you add test cases about validation?

**Comment by Josh-Cena** [comment]
Very nice! Appreciate the amazing work ❤️

**Comment by christopherklint97** [comment]
Thank you for all the great help! 😄 💯

File: packages/docusaurus-theme-classic/src/__tests__/validateThemeConfig.test.js [hunk 1]
```diff
@@ -345,6 +345,68 @@ describe('themeConfig', () => {
...
+            href: 'https://opensource.facebook.com/legal/privacy/',
+          },
+        ],
+      },
+    };
+
+    expect(() =>
+      testValidateThemeConfig(partialConfig),
+    ).toThrowErrorMatchingInlineSnapshot(
+      `"\\"footer.links\\" does not match any of the allowed types"`,
```
Comment (Josh-Cena) [inline packages/docusaurus-theme-classic/src/__tests__/validateThemeConfig.test.js:406]: The message doesn't seem so friendly to me. I know custom messages are very hard in Joi, but could you try to improve that to "the footer must either be simple or multi-column, but not a mix, see https://docusaurus.io/docs/api/themes/configuration#footer-links", or something like that?

Comment (christopherklint97) [reply packages/docusaurus-theme-classic/src/__tests__/validateThemeConfig.test.js]: Completely agree! Should be fixed now.

**Comment by Josh-Cena** [comment]
Thanks again for the work!

**Comment by Josh-Cena** [comment]
@slorber you may want to test this locally and check the design before merging

**Review by slorber — APPROVED** [review]
Thanks

Did some little modifications but overall that looks good.

For now I kept the simple footer centered above the copyright

**Pull request merged**